src/app/replayer/replayer.ml

```diff
@@ -728,6 +700,729 @@ let run_user_command ~logger ~pool ~ledger (cmd : Sql.User_command.t)
+                      Snapp_helpers.state_data_of_ids ~pool snapp_state_ids
+                    in
+                    Array.map state_data ~f:Snapp_basic.Or_ignore.of_option
+                    |> Array.to_list
+                    |> Pickles_types.Vector.Vector_8.of_list_exn
+                  in
+                  let%bind sequence_state =
+                    Option.value_map snapp_account_data.sequence_state_id
+                      ~default:(return Snapp_basic.Or_ignore.Ignore)
+                      ~f:(fun state_id ->
+                        let%map state_data_str =
+                          query_db pool ~item:"Snapp state data" ~f:(fun db ->
+                              Processor.Snapp_state_data.load db state_id)
+                        in
+                        let state_data =
+                          Pickles.Backend.Tick.Field.of_string state_data_str
+                        in
+                        Snapp_basic.Or_ignore.Check state_data)
+                  in
+                  let proved_state =
+                    Option.value_map snapp_account_data.proved_state
+                      ~default:Snapp_basic.Or_ignore.Ignore ~f:(fun b ->
+                        Snapp_basic.Or_ignore.Check b)
+                  in
+                  return
+                    ( { balance
+                      ; nonce
+                      ; receipt_chain_hash
+                      ; public_key
+                      ; delegate
+                      ; state
+                      ; sequence_state
+                      ; proved_state
+                      }
+                      : Snapp_predicate.Account.t )
+                in
+                Party.Predicate.Full snapp_account
+            | Nonce -> (
+                match predicate_data.nonce with
+                | None ->
+                    failwith "Expected nonce for predicate of kind Nonce"
+                | Some nonce ->
+                    return
+                      (Party.Predicate.Nonce
+                         (Mina_numbers.Account_nonce.of_uint32
+                            (Unsigned.UInt32.of_int64 nonce))) )
+            | Accept ->
+                return Party.Predicate.Accept
+          in
+          return ({ body; predicate } : Party.Predicated.t)
+        in
+        let authorization =
+          (* dummy proof, signature, don't affect replay *)
+          match snapp_party_data.authorization_kind with
+          | Control.Tag.Proof ->
+              let n2 = Pickles_types.Nat.N2.n in
+              let proof = Pickles.Proof.dummy n2 n2 n2 in
+              Control.Proof proof
+          | Control.Tag.Signature ->
+              Control.Signature Signature.dummy
+          | Control.Tag.None_given ->
+              Control.None_given
+        in
+        return ({ data; authorization } : Party.t))
+  in
+  (* memo contents don't affect ability to replay snapp *)
+  let memo = Mina_base.Signed_command_memo.dummy in
```

Comment:
We should add memo to the archive database though.

Reply:
We haven't been doing that for ordinary transactions. Should we add that for Snapps? For all transactions?

Reply:
Hmm, I think we should? cc @bkase